erpnext/controllers/taxes_and_totals.py

```diff
@@ -477,25 +484,26 @@ def get_current_tax_and_net_amount(self, item, tax, item_tax_map):
 		current_net_amount = 0.0
 
 		if tax.charge_type == "Actual":
+			current_net_amount = item.net_amount
 			# distribute the tax amount proportionally to each item row
 			actual = flt(tax.tax_amount, tax.precision("tax_amount"))
 
 			if tax.get("is_tax_withholding_account") and item.meta.get_field("apply_tds"):
 				if not item.get("apply_tds") or not self.doc.tax_withholding_net_total:
 					current_tax_amount = 0.0
-					current_net_amount = 0.0
 				else:
-					current_net_amount = item.net_amount
-					current_tax_amount = current_net_amount * actual / self.doc.tax_withholding_net_total
+					current_tax_amount = item.net_amount * actual / self.doc.tax_withholding_net_total
 			else:
-				current_net_amount = item.net_amount
 				current_tax_amount = (
-					current_net_amount * actual / self.doc.net_total if self.doc.net_total else 0.0
+					item.net_amount * actual / self.doc.net_total if self.doc.net_total else 0.0
 				)
 
 		elif tax.charge_type == "On Net Total":
-			current_net_amount = item.net_amount
-			current_tax_amount = (tax_rate / 100.0) * current_net_amount
+			if not item_tax_map:
+				current_net_amount = item.net_amount
```

Comment:
This is not correct and also sums net_amounts that have been caused by a document-wide tax of type "On Net Total" which is characterized by not having been been overridden by an item tax template and thus not bearing an `item_tax_map`

Reply:
Proper fix will be preceded by a series of clarifications, such as https://github.com/frappe/erpnext/pull/44226